File: regexlib/2021-5-15/javascript_test_file/regexlib_1524.js
/* 1524
 * \[(?<GroupName>.*)\](?<GroupContent>[^\[]+)       --------        [\s]*(?<Key>.+)[\s]*=[\s]*(?<Value>[^\r]+) 
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"[]-       --------        1"*256+"!1 _SLQ_2"
 */
var REGEX = /\[(?<GroupName>.*)\](?<GroupContent>[^\[]+)       --------        [\s]*(?<Key>.+)[\s]*=[\s]*(?<Value>[^\r]+) /
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '[]-       --------        1'.repeat(i*1) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/2021-5-15/python_re2_test_file/regexlib_1524.py
# 1524
# \[(?<GroupName>.*)\](?<GroupContent>[^\[]+)       --------        [\s]*(?<Key>.+)[\s]*=[\s]*(?<Value>[^\r]+) 
# EXPONENT
# nums:5
# EXPONENT AttackString:""+"[]-       --------        1"*256+"!1 _SLQ_2"

import re2 as re
from time import perf_counter

regex = """\[(?<GroupName>.*)\](?<GroupContent>[^\[]+)       --------        [\s]*(?<Key>.+)[\s]*=[\s]*(?<Value>[^\r]+) """
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "[]-       --------        1" * i * 1 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")